std/experimental/allocator/building_blocks/ascending_page_allocator.d

```diff
@@ -100,7 +102,7 @@ public:
         import std.algorithm.comparison : min;
 
         size_t goodSize = goodAllocSize(n);
-        if (offset - data > numPages * pageSize - goodSize)
+        if (goodSize > numPages * pageSize || offset - data > numPages * pageSize - goodSize)
```

Comment:
You may want to introduce `immutable pagedBytes = numPages * pageSize;` because it's used a number of times throughout the function,